cmd/api/src/services/fileupload/file_upload.go

```diff
@@ -147,6 +131,24 @@ func SaveIngestFile(location string, request *http.Request) (string, model.FileT
 	}
 }
 
+type FileValidator func(src io.Reader, dst io.Writer) error
```

Comment:
If the WriteAndValidateFile function returned the filename, modeType, and error you'd be able to eliminate the if / else checks above as well correct?